frontend/src/metabase/admin/tasks/containers/TaskModal/TaskModal.tsx

```diff
@@ -0,0 +1,89 @@
+import { useMemo } from "react";
+import { Link } from "react-router";
+import { goBack } from "react-router-redux";
+import { t } from "ttag";
+
+import { useGetTaskQuery } from "metabase/api";
+import { CodeBlock } from "metabase/components/CodeBlock";
+import { CopyButton } from "metabase/components/CopyButton";
+import { LoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper";
+import ModalContent from "metabase/components/ModalContent";
+import { openSaveDialog } from "metabase/lib/dom";
+import { useDispatch } from "metabase/lib/redux";
+import { Box, Button, Flex, Icon } from "metabase/ui";
+import type { Task } from "metabase-types/api";
+
+import S from "./TaskModal.module.css";
+
+interface Props {
+  params: { taskId: number };
+}
+
+export const TaskModal = ({ params }: Props) => {
+  const dispatch = useDispatch();
+  const { data: task, error, isLoading } = useGetTaskQuery(params.taskId);
+  const code = useMemo(() => formatTaskDetails(task), [task]);
+  const linesCount = useMemo(() => code.split("\n").length, [code]);
+
+  const handleClose = () => {
+    dispatch(goBack());
+  };
+
+  const handleDownload = () => {
+    const filename = getFilename(task);
+    const blob = new Blob([code], { type: "text/json" });
+    openSaveDialog(filename, blob);
+  };
+
+  if (error || isLoading) {
+    return <LoadingAndErrorWrapper error={error} loading={isLoading} />;
+  }
+
+  return (
+    <ModalContent title={t`Task details`} onClose={handleClose}>
+      <Box
+        className={S.codeContainer}
+        p={linesCount > 1 ? 0 : "xs"}
+        pos="relative"
+      >
+        <CodeBlock
+          basicSetup={{
+            /**
+             * Hide line numbers when there's only one line to avoid
+             * confusion about whether it's part of the log output.
+             */
+            lineNumbers: linesCount > 1,
```

Comment:
With this:

![image](https://github.com/user-attachments/assets/5bdc8635-02c0-4bb7-9317-d9f2733a965c)

With line numbers shown:

![image](https://github.com/user-attachments/assets/f804084a-019e-426d-a51e-b5fc5ed27e3d)